violinplot.m

```diff
@@ -104,24 +119,28 @@
             thisData = data(cats == thisCat);
             violins(n) = Violin(thisData, n, varargin{:});
         end
-        set(gca, 'XTick', 1:length(catnames), 'XTickLabels', catnames_labels);
+        set(AxisHandle, 'XTick', 1:length(catnames), 'XTickLabel', catnames_labels);
 
     % 1D data, no categories
     elseif not(hascategories) && isvector(data)
         violins = Violin(data, 1, varargin{:});
-        set(gca, 'XTick', 1);
+        set(AxisHandle, 'XTick', 1);
 
     % 2D data with or without categories
     elseif ismatrix(data)
         for n=1:size(data, 2)
             thisData = data(:, n);
             violins(n) = Violin(thisData, n, varargin{:});
         end
-        set(gca, 'XTick', 1:size(data, 2));
+        set(AxisHandle, 'XTick', 1:size(data, 2));
         if hascategories && length(cats) == size(data, 2)
-            set(gca, 'XTickLabels', cats);
+            set(AxisHandle, 'XTickLabel', cats);
         end
 
     end
+    
+    if nargout > 0
+        varargout{1} = violins;
```

Comment:
Why the change to varargout?

Reply:
Sometimes I run the function from the command line, and I don't want any output. This gives it the flexibility for that option in addition to the default output.

Reply:
That doesn't make sense to me. If you don't assign the return value, you don't receive it. Please revert this part of the change before merging.

Reply:
That was the point of this change. When I am running the function from the command line, sometimes I do not want to receive the return value.

Reply:
Please revert this change. If you do not want to receive the return value, put a semicolon at the end of the line.